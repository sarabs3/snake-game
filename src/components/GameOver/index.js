import React from 'react';

const GameOver = props => {
	return (
		<div className="snake-app__overlay">
			<div className="mb-1" data-testid={'game-over-message'}><b>GAME OVER!</b></div>
			<div className="mb-1" data-testid={'game-over-score'}>Your score: {props.score} </div>
			<button onClick={props.start} data-testid={'start-game-button'}>Start a new game</button>
		</div>
	);
};

export default GameOver;
