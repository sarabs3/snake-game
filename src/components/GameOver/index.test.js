import React from 'react';
import ReactDOM from 'react-dom';

import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react';

import GameOver from './index';

let div;

beforeEach(() => {
	div = document.createElement('div');
	document.body.appendChild(div);
});

afterEach(() => {
	ReactDOM.unmountComponentAtNode(div);
	div.remove();
	div = null;
});

const startGame = jest.fn();

it('<GameOver /> renders without crashing', () => {
	const {queryByTestId} = render(<GameOver score={10} start={startGame} />, div);

	expect(queryByTestId('game-over-message')).toBeTruthy();
	expect(queryByTestId('game-over-score')).toBeTruthy();
	expect(queryByTestId('start-game-button')).toBeTruthy();
});

it ('<GameOver /> snapshot matched', () => {
	const tree = renderer.create(<GameOver score={10} start={startGame} />, div).toJSON();

	expect(tree).toMatchSnapshot();
});

it('<GameOver /> start game button working', () => {
	const {queryByTestId} = render(<GameOver score={10} start={startGame} />, div);

	const button = queryByTestId('start-game-button');

	expect(button.textContent).toBe('Start a new game');

	fireEvent.click(button);

	expect(startGame).toHaveBeenCalledTimes(1);

	for (let i = 0; i < 5; i ++) {
		fireEvent.click(button);
	}

	expect(startGame).toHaveBeenCalledTimes(6);
});
