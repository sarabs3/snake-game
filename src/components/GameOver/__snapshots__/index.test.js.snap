// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOver /> snapshot matched 1`] = `
<div
  className="snake-app__overlay"
>
  <div
    className="mb-1"
    data-testid="game-over-message"
  >
    <b>
      GAME OVER!
    </b>
  </div>
  <div
    className="mb-1"
    data-testid="game-over-score"
  >
    Your score: 
    10
     
  </div>
  <button
    data-testid="start-game-button"
    onClick={[MockFunction]}
  >
    Start a new game
  </button>
</div>
`;
